packages/js/src/novu.ts

```diff
@@ -30,13 +33,17 @@ export class Novu implements Pick<NovuEventEmitter, 'on' | 'off'> {
     this.#session.initialize();
     this.feeds = new Feeds();
     this.preferences = new Preferences();
+    this.#socket = new Socket({ socketUrl: options.socketUrl });
   }
 
   on<Key extends EventNames>(eventName: Key, listener: EventHandler<Events[Key]>): void {
+    if (this.#socket.isSocketEvent(eventName)) {
+      this.#socket.initialize();
+    }
```

Comment:
When someone tries to listen to one of the WS events then initialize the WebSocket connection and add listeners on events.